### ISSUE TITLE
Unable to generate Blood Pressure values with customised hypertension module

### ISSUE THREAD

**Issue by skviswa** [synthetichealth/synthea#824]
I have been trying to generate some synthetic data for hypertension indication. 
I reviewed the default Synthea module, and I found that I needed to make slight customisations, such as adding a value range for the Systolic and Diastolic blood pressure records to generated, and setting hypertension attribute so that it is able to traverse the module. Not sure if it is possible to attach the json file of the customised module here, I will try adding it as a comment.
When I generate data, I see that I am generating many of the other states except this Observation record. For reference, I have been trying to generate FHIR DSTU2 resources. I can see for example records with the hypertension condition diagnosis, medications such as Hydrochlorothiazide, Atenolol etc generated. 
Is there something that I have missed here or is there a bug with MultiObservation module?

**Comment by skviswa** [comment]
`{
  "name": "hypertension_with_value",
  "remarks": [
    "Hypertension modelled after the JNC 8 Hypertension Guideline Algorithm.",
    "",
    "The hypertension model section located in the beginning of the Metabolic Syndrome Standard of Care   ",
    "",
    "The recommendation for first-line therapy for hypertension remains a beta blocker or diuretic given in a low dosage. A target blood pressure of less than 140/90 mm Hg is achieved in about 50 percent of patients treated with monotherapy; two or more agents from different pharmacologic classes are often needed to achieve adequate blood pressure control. Single-dose combination antihypertension therapy is an important option that combines efficacy of blood pressure reduction and a low side effect profile with convenient once-daily dosing to enhance compliance. Combination antihypertensives include combined agents from the following pharmacologic classes: diuretics and potassium-sparing diuretics, beta blockers and diuretics, angiotensin-converting enzyme (ACE) inhibitors and diuretics, angiotensin-II antagonists and diuretics, and calcium channel blockers and ACE inhibitors - https://www.aafp.org/afp/2000/0515/p3049.html"
  ],
  "states": {
    "Initial": {
      "type": "Initial",
      "remarks": [
        "Initial impl == direct translation of ruby module"
      ],
      "direct_transition": "Initial_Hypertension_with_value_Status"
    },
    "Initial_Hypertension_with_value_Status": {
      "type": "SetAttribute",
      "attribute": "hypertension",
      "direct_transition": "Initial_Hypertension_with_value_Status_2",
      "value": true
    },
    "Initial_Hypertension_with_value_Status_2": {
      "type": "SetAttribute",
      "attribute": "hypertension_dx",
      "direct_transition": "Wellness_Encounter_with_value"
    },
    "Wellness_Encounter_with_value": {
      "type": "Encounter",
      "wellness": true,
      "conditional_transition": [
        {
          "transition": "Diagnose_Hypertension_with_value",
          "condition": {
            "condition_type": "And",
            "conditions": [
              {
                "condition_type": "Attribute",
                "attribute": "hypertension",
                "operator": "==",
                "value": true
              },
              {
                "condition_type": "Attribute",
                "attribute": "hypertension_dx",
                "operator": "is nil"
              }
            ]
          }
        },
        {
          "transition": "End_Wellness_No_Action_with_value"
        }
      ]
    },
    "End_Wellness_No_Action_with_value": {
      "type": "EncounterEnd",
      "direct_transition": "Wellness_Encounter_with_value"
    },
    "Diagnose_Hypertension_with_value": {
      "type": "ConditionOnset",
      "codes": [
        {
          "system": "SNOMED-CT",
          "code": 59621000,
          "display": "Hypertension"
        }
      ],
      "assign_to_attribute": "hypertension_dx",
      "direct_transition": "Set_BP_Not Controlled_with_value"
    },
    "Set_BP_Not Controlled_with_value": {
      "type": "SetAttribute",
      "attribute": "blood_pressure_controlled",
      "direct_transition": "LifeStyle_Modifications_Hypertension_with_value_CarePlan",
      "value": false
    },
    "LifeStyle_Modifications_Hypertension_with_value_CarePlan": {
      "type": "CarePlanStart",
      "codes": [
        {
          "system": "SNOMED-CT",
          "code": 443402002,
          "display": "Lifestyle education regarding hypertension"
        }
      ],
      "activities": [
        {
          "system": "SNOMED-CT",
          "code": 386463000,
          "display": "Prescribed activity/exercise education"
        },
        {
          "system": "SNOMED-CT",
          "code": 413473000,
          "display": "Counseling about alcohol consumption"
        },
        {
          "system": "SNOMED-CT",
          "code": 1151000175103,
          "display": "Dietary approaches to stop hypertension diet"
        },
        {
          "system": "SNOMED-CT",
          "code": 225323000,
          "display": "Smoking cessation education"
        }
      ],
      "goals": [
        {
          "addresses": [
            "hypertension_dx"
          ],
          "text": "Maintain blood pressure below 140/90 mm[Hg]"
        },
        {
          "addresses": [
            "hypertension_dx"
          ],
          "text": "Reduce sodium intake to no more than 2,400 mg/day"
        }
      ],
      "reason": "hypertension_dx",
      "conditional_transition": [
        {
          "transition": "Hypertension_with_value_Medication_Low",
          "condition": {
            "condition_type": "Vital Sign",
            "vital_sign": "Systolic Blood Pressure",
            "operator": "<",
            "value": 160
          }
        },
        {
          "transition": "Hypertension_with_value_Medication_High",
          "condition": {
            "condition_type": "Vital Sign",
            "vital_sign": "Systolic Blood Pressure",
            "operator": ">=",
            "value": 160
          }
        }
      ]
    },
    "Hypertension_with_value_Medication_Low": {
      "type": "MedicationOrder",
      "codes": [
        {
          "system": "RxNorm",
          "code": 429503,
          "display": "Hydrochlorothiazide 12.5 MG"
        }
      ],
      "prescription": {
        "dosage": {
          "amount": 1,
          "frequency": 1,
          "period": 1,
          "unit": "days"
        }
      },
      "chronic": true,
      "direct_transition": "End_Wellness_Encounter_with_value",
      "assign_to_attribute": "hypertension_medication",
      "reason": "hypertension_dx"
    },
    "Hypertension_with_value_Medication_High": {
      "type": "MedicationOrder",
      "codes": [
        {
          "system": "RxNorm",
          "code": 310798,
          "display": "Hydrochlorothiazide 25 MG Oral Tablet"
        }
      ],
      "prescription": {
        "dosage": {
          "amount": 1,
          "frequency": 1,
          "period": 1,
          "unit": "days"
        }
      },
      "chronic": true,
      "direct_transition": "End_Wellness_Encounter_with_value",
      "assign_to_attribute": "hypertension_medication",
      "reason": "hypertension_dx"
    },
    "End_Wellness_Encounter_with_value": {
      "type": "EncounterEnd",
      "distributed_transition": [
        {
          "transition": "Set_BP_Controlled_with_value",
          "distribution": 0.49
        },
        {
          "transition": "Delay_One_Month_with_value",
          "distribution": 0.51
        }
      ]
    },
    "Set_BP_Controlled_with_value": {
      "type": "SetAttribute",
      "attribute": "blood_pressure_controlled",
      "value": true,
      "direct_transition": "Delay_One_Month_with_value"
    },
    "Delay_One_Month_with_value": {
      "type": "Delay",
      "exact": {
        "quantity": 1,
        "unit": "months"
      },
      "direct_transition": "Hypertension_with_value_Followup_Encounter"
    },
    "Hypertension_with_value_Followup_Encounter": {
      "type": "Encounter",
      "encounter_class": "ambulatory",
      "reason": "",
      "codes": [
        {
          "system": "SNOMED-CT",
          "code": 390906007,
          "display": "Hypertension follow-up encounter"
        }
      ],
      "direct_transition": "Record_BP_with_value"
    },
    "Record_BP_with_value": {
      "type": "MultiObservation",
      "category": "vital-signs",
      "codes": [
        {
          "system": "LOINC",
          "code": "85354-9",
          "display": "Blood Pressure"
        }
      ],
      "observations": [
        {
          "category": "vital-signs",
          "codes": [
            {
              "system": "LOINC",
              "code": "8480-6",
              "display": "Systolic Blood Pressure"
            }
          ],
          "unit": "mm[Hg]",
          "range": {
            "low": 130,
            "high": 160
          }
        },
        {
          "category": "vital-signs",
          "codes": [
            {
              "system": "LOINC",
              "code": "8462-4",
              "display": "Diastolic Blood Pressure"
            }
          ],
          "unit": "mm[Hg]",
          "range": {
            "low": 80,
            "high": 100
          }
        }
      ],
      "conditional_transition": [
        {
          "transition": "End_Hypertension_with_value_Medication",
          "condition": {
            "condition_type": "Attribute",
            "attribute": "blood_pressure_controlled",
            "operator": "==",
            "value": false
          }
        },
        {
          "transition": "End_Hypertension_with_value_Followup_Encounter"
        }
      ]
    },
    "End_Hypertension_with_value_Medication": {
      "type": "MedicationEnd",
      "referenced_by_attribute": "hypertension_medication",
      "direct_transition": "Hypertension_with_value_Combination_Medication"
    },
    "Hypertension_with_value_Combination_Medication": {
      "type": "MedicationOrder",
      "codes": [
        {
          "system": "RxNorm",
          "code": 746030,
          "display": "Atenolol 50 MG / Chlorthalidone 25 MG Oral Tablet"
        }
      ],
      "prescription": {
        "dosage": {
          "amount": 1,
          "frequency": 1,
          "period": 1,
          "unit": "days"
        }
      },
      "chronic": true,
      "assign_to_attribute": "hypertension_medication_2",
      "reason": "hypertension_dx",
      "direct_transition": "End_Hypertension_with_value_Followup_Encounter"
    },
    "End_Hypertension_with_value_Followup_Encounter": {
      "type": "EncounterEnd",
      "complex_transition": [
        {
          "condition": {
            "condition_type": "Attribute",
            "attribute": "blood_pressure_controlled",
            "operator": "==",
            "value": false
          },
          "distributions": [
            {
              "transition": "Set_BP_with_value_Controlled_2",
              "distribution": 0.49
            },
            {
              "transition": "Delay 2_Month_with_value",
              "distribution": 0.51
            }
          ]
        },
        {
          "distributions": [],
          "transition": "Wellness_Encounter_with_value"
        }
      ]
    },
    "Set_BP_with_value_Controlled_2": {
      "type": "SetAttribute",
      "attribute": "blood_pressure_controlled",
      "value": true,
      "direct_transition": "Delay 2_Month_with_value"
    },
    "Delay 2_Month_with_value": {
      "type": "Delay",
      "exact": {
        "quantity": 2,
        "unit": "months"
      },
      "direct_transition": "Hypertension_with_value_Followup_Encounter_2"
    },
    "Hypertension_with_value_Followup_Encounter_2": {
      "type": "Encounter",
      "encounter_class": "ambulatory",
      "reason": "",
      "codes": [
        {
          "system": "SNOMED-CT",
          "code": 390906007,
          "display": "Hypertension follow-up encounter"
        }
      ],
      "direct_transition": "Record_BP_with_value_2"
    },
    "Record_BP_with_value_2": {
      "type": "MultiObservation",
      "category": "vital-signs",
      "codes": [
        {
          "system": "LOINC",
          "code": "85354-9",
          "display": "Blood Pressure"
        }
      ],
      "observations": [
        {
          "category": "vital-signs",
          "codes": [
            {
              "system": "LOINC",
              "code": "8480-6",
              "display": "Systolic Blood Pressure"
            }
          ],
          "unit": "mm[Hg]",
          "range": {
            "low": 120,
            "high": 150
          }
        },
        {
          "category": "vital-signs",
          "codes": [
            {
              "system": "LOINC",
              "code": "8462-4",
              "display": "Diastolic Blood Pressure"
            }
          ],
          "unit": "mm[Hg]",
          "range": {
            "low": 80,
            "high": 89
          }
        }
      ],
      "conditional_transition": [
        {
          "transition": "End_Hypertension_with_value_Comb_Medication",
          "condition": {
            "condition_type": "Attribute",
            "attribute": "blood_pressure_controlled",
            "operator": "==",
            "value": false
          }
        },
        {
          "transition": "End_Hypertension_with_value_Followup_Encounter_2"
        }
      ]
    },
    "End_Hypertension_with_value_Comb_Medication": {
      "type": "MedicationEnd",
      "referenced_by_attribute": "hypertension_medication_2",
      "direct_transition": "Hypertension_Combination_with_value_Medication_2"
    },
    "Hypertension_Combination_with_value_Medication_2": {
      "type": "MedicationOrder",
      "codes": [
        {
          "system": "RxNorm",
          "code": 999967,
          "display": "amLODIPine 5 MG / Hydrochlorothiazide 12.5 MG / Olmesartan medoxomil 20 MG Oral Tablet"
        }
      ],
      "prescription": {
        "dosage": {
          "amount": 1,
          "frequency": 1,
          "period": 1,
          "unit": "days"
        }
      },
      "chronic": true,
      "assign_to_attribute": "hypertension_medication_3",
      "reason": "hypertension_dx",
      "direct_transition": "End_Hypertension_with_value_Followup_Encounter_2",
      "remarks": [
        "Olmesartan is an angiotensin II receptor blocker (ARB) that relaxes the blood vessels.",
        "Amlodipine is a calcium channel blocker that relaxes the blood vessels.",
        "Hydrochlorothiazide is a thiazide diuretic (water pill). It is used to help reduce the amount of water in the body by increasing the flow of urine."
      ]
    },
    "End_Hypertension_with_value_Followup_Encounter_2": {
      "type": "EncounterEnd",
      "complex_transition": [
        {
          "condition": {
            "condition_type": "Attribute",
            "attribute": "blood_pressure_controlled",
            "operator": "==",
            "value": false
          },
          "distributions": [
            {
              "transition": "Set_BP_with_value_Controlled_3",
              "distribution": 0.49
            },
            {
              "transition": "Delay_2_Month_2_with_value",
              "distribution": 0.51
            }
          ]
        },
        {
          "distributions": [],
          "transition": "Wellness_Encounter_with_value"
        }
      ]
    },
    "Set_BP_with_value_Controlled_3": {
      "type": "SetAttribute",
      "attribute": "blood_pressure_controlled",
      "value": true,
      "direct_transition": "Delay_2_Month_2_with_value"
    },
    "Delay_2_Month_2_with_value": {
      "type": "Delay",
      "exact": {
        "quantity": 2,
        "unit": "months"
      },
      "direct_transition": "Hypertension_with_value_Followup_Encounter_3"
    },
    "Hypertension_with_value_Followup_Encounter_3": {
      "type": "Encounter",
      "encounter_class": "ambulatory",
      "reason": "",
      "codes": [
        {
          "system": "SNOMED-CT",
          "code": 390906007,
          "display": "Hypertension follow-up encounter"
        }
      ],
      "direct_transition": "Record_BP_with_value_3"
    },
    "Record_BP_with_value_3": {
      "type": "MultiObservation",
      "category": "vital-signs",
      "codes": [
        {
          "system": "LOINC",
          "code": "85354-9",
          "display": "Blood Pressure"
        }
      ],
      "observations": [
        {
          "category": "vital-signs",
          "codes": [
            {
              "system": "LOINC",
              "code": "8480-6",
              "display": "Systolic Blood Pressure"
            }
          ],
          "unit": "mm[Hg]",
          "range": {
            "low": 130,
            "high": 139
          }
        },
        {
          "category": "vital-signs",
          "codes": [
            {
              "system": "LOINC",
              "code": "8462-4",
              "display": "Diastolic Blood Pressure"
            }
          ],
          "unit": "mm[Hg]",
          "range": {
            "low": 80,
            "high": 89
          }
        }
      ],
      "conditional_transition": [
        {
          "transition": "Referral To Hypertension_with_value Clinic",
          "condition": {
            "condition_type": "Attribute",
            "attribute": "blood_pressure_controlled",
            "operator": "==",
            "value": false
          }
        },
        {
          "transition": "End_Hypertension_with_value_Followup_Encounter_3"
        }
      ]
    },
    "Referral To Hypertension_with_value Clinic": {
      "type": "Procedure",
      "codes": [
        {
          "system": "SNOMED-CT",
          "code": 183856001,
          "display": "Referral to hypertension clinic"
        }
      ],
      "direct_transition": "End_Hypertension_with_value_Followup_Encounter_3"
    },
    "End_Hypertension_with_value_Followup_Encounter_3": {
      "type": "EncounterEnd",
      "direct_transition": "Wellness_Encounter_with_value"
    }
  }
}`

**Comment by dehall** [comment]
I think there are 2 things going on here:
1. I'm guessing that you're running Synthea using the `-m hypertension` flag, in order to get hypertension patients. One issue is that the hypertension module actually depends on the "Metabolic Syndrome Disease Progression" module to set hypertension = true. It's also important to point out that Synthea runs all modules by default, and the `-m` flag is meant to exclude the rest of the modules. It doesn't make patients more likely to have the condition in question.
To quote from a previous issue:
> This is the problem with the entire module filter option. If you don't specifically include it, it is not included. Which is why for most people we do not recommend it be used, unless you want to take the time to understand the connections between all the modules and include the dependencies.  

So in short, I suggest trying again without using the `-m` flag.

2. By default, Synthea only exports the last 10 years of patient history, and only "active" entries prior to that. So for example an active diagnosis of hypertension from 20 years ago will be kept, but the observation of blood pressure taken at diagnosis will be filtered out.  You can either disable the filter, or generate more patients until you find one with the right data happening within the last 10 years. 
The filter can be disabled by opening the file `src/main/resources/synthea.properties`, and changing the `exporter.years_of_history` setting to `0` instead of 10 - this will keep every patient's entire record from birth until death/the present day.

If those 2 changes don't solve your issue, we're going to need more information about how you're running Synthea and the results you're getting. Unless there's some edge case I'm not seeing, the blood pressure MultiObservations should be written in all FHIR versions.

**Comment by skviswa** [comment]
@dehall thank you for the inputs! I will try those suggestions and get back to you. For better context, I generated the data by using the -m flag and passed the name of the custom module `hypertension_with_value`. I did notice that the hypertension default module was just checking the attribute so in my modifications I set the attribute using the `SetAttribute` state. 
And with regards to years of history, I was trying for last 5 years actually instead of the default 10, as I needed only the last 5 years of history.
I will try the changes you suggested and get back on this thread if that solves my issue.
